services/dataAccessLayer.js

```diff
@@ -70,11 +91,44 @@ const removeSensitiveInfo = function (obj) {
   }
 };
 
+const privilegedAccess = (user, data, level) => {
+  if ("email" in data) {
+    user.email = data.email;
+  }
+  if (level === ACCESS_LEVEL.PRIVATE || level === ACCESS_LEVEL.CONFIDENTIAL) {
+    if ("phone" in data) {
+      user.phone = data.phone;
+    }
+  }
+  if (level === ACCESS_LEVEL.CONFIDENTIAL) {
+    if ("chaincode" in data) {
+      user.chaincode = data.chaincode;
+    }
+  }
+  return user;
+};
+
+const levelSpecificAccess = (user, level = ACCESS_LEVEL.PUBLIC, role = null) => {
+  const unFilteredData = JSON.parse(JSON.stringify(user));
```

Comment:
Isn't this negating each other?

Reply:
you are stringifying and parsing it back again. both these operations will nullify each other.